test/realm_test.dart

```diff
@@ -552,5 +552,161 @@ Future<void> main([List<String>? args]) async {
       expect(() => realm.write(() => players[-1] = Person()), throws<RealmException>("Index out of range"));
       expect(() => realm.write(() => players[800] = Person()), throws<RealmException>());
     });
+
+    test('RealmList clear items from list', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create a team
+      final team = Team()..name = "Team";
+      realm.write(() => realm.add(team));
+
+      //Add players to the team
+      final newPlayers = [
+        Person()..name = "Michael Schumacher",
+        Person()..name = "Sebastian Vettel",
+        Person()..name = "Kimi Räikkönen",
+      ];
+      realm.write(() {
+        team.players.addAll(newPlayers);
+      });
+
+      //Ensure teams and players are in realm
+      var teams = realm.all<Team>();
+      expect(teams.length, 1);
```

Comment:
We don't need to do this. Any test should try to test only the things it needs to test. If `realm.write` fails there should be a specific test for that. And we should not test if `write` works here.

Reply:
scratch that. I see the point